samples/ipc_call_interface/producer/math_1_producer_1.c

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include "az_ulib_result.h"
+#include "interfaces/math_1_interface.h"
+#include <stdio.h>
+
+void math_1_producer_1_create(void)
+{
+  (void)printf("Create producer v1 for math v1 ...\r\n");
+  az_result result;
+  if ((result = math_1_publish_interface()) != AZ_OK)
+  {
+    (void)printf("Producer v1 publish math v1 interface failed with error %d\r\n", result);
+  }
+  else
+  {
+    (void)printf("Producer v1 publish math v1 interface with success\r\n");
```

Comment:
This feels backwards from the filename. Should the file be `producer_1_math_1.c`?

Reply:
Can the producer idea have multiple versions separate from versioning of an interface?

Reply:
Yes, interface version is independent of the producer version.

---
In reply to: [572480963](https://github.com/Azure/azure-ulib-c/pull/60#discussion_r572480963) [](ancestors = 572480963)